futures-util/Cargo.toml

```diff
@@ -29,6 +30,7 @@ futures-io-preview = { path = "../futures-io", version = "0.3.0-alpha.1", defaul
 futures-sink-preview = { path = "../futures-sink", version = "0.3.0-alpha.1", default-features = false}
 either = { version = "1.4", default-features = false }
 slab = { version = "0.4", optional = true }
+futures = { version = "0.1", optional = true }
```

Comment:
I’m surprised this plus the dev-dependency below don’t cause issues for the tests (both have a lib name of `futures`). Again here’s a place where having crate renaming for the edition would be super useful to make this `futures01` or something.

Reply:
We should convene the council of elders

One temporary solution *could* be to fully rename our libraries to `futures-preview`. Long term though, we'll need crate renaming because both crates will share a name in the end.

Reply:
could we get away with moving `Compat` into its own `futures-compat` sub crate?

Reply:
I think that'll work for now, since it doesn't need to import the 0.3 `futures` facade it won't run into the name conflict. Once renaming is working then it could be merged back into `futures-util`.

Reply:
@tinaun We might. It only happens in the doc tests because `futures-util` has `futures-preview` as dev dependency. So the normal dep `futures@0.1` and dev dep collide.

I think it is problematic that one can't depend on `futures` 0.3 and 0.1 in the same project at the same time

Reply:
Alright, testing with a locally built `cargo` including https://github.com/rust-lang/cargo/pull/5794 using `futures01 = { version = "0.1", optional = true, package = "futures" }` (and adjusting all the `use`s) appears to run the tests fine.

So, if you're willing to wait for a new `cargo` to be released to nightly that's one option.

Reply:
@Nemo157 How long do you think will this take? If it's just a day or two, then we should wait. @tinaun requires some time to tweak this PR anyway

Reply:
I'm not sure what the Cargo release cycle is like, my current nightly toolchain has `rustc` from 2018-07-23 but `cargo` from 2018-07-17, so I don't think it's always the latest master.

Reply:
this pr requires a new nightly anyway for the Executor stuff

Reply:
@tinaun Wouldn't the newtype approach be more lightweight?